decisions/0003-blah.md

```diff
@@ -0,0 +1,57 @@
+# Title
+
+Date: 2022-06-01
+
+Status: proposed
+
+## Context
+
+Currently, users need to install 4 Remix dependencies:
+
+- Dev (`@remix-run/dev`)
+- Server runtime (`@remix-run/{cloudflare,deno,node}`)
+- Adapter (`@remix-run/{arc,cloudflare-pages,cloudflare-workers,express,fly,netlify,vercel}`)
+- Renderer (`@remix-run/react`)
+
+Code duplication, docs complexity, unintuitive for users.
+Bugs (link to PR fixes where we forgot to export things, enumerate discrepancies)
+Plus, you can import something like `createFileSessionStorage` from `@remix-run/node`, but that won't work on some platforms like `netlify` or `vercel`.
+
+```tsx
+import { createFileSessionStorage } from "@remix-run/node";
+// Implementation only depends on server runtime, but _availability_ depends on the adapter!
+// This won't work in netlify nor in vercel!
+```
+
+## Decision
+
+Users should import from _adapter packages_, not from runtime packages.
+
+### user facing
+- `@remix-run/dev` -> `remix`, `@remix-run/cli`
+- `@remix-run/react`
+- `@remix-run/server`
+- Adapters: `@remix-run/{adapter}` OR `@remix-run/serve`
+
+### internal
+- Runtimes: `@remix-run/{cloudflare,deno,node}`
+- `@remix-run/shared`?
+
+## Consequences
+
+
+---
+
+Key insight: _adapters_ have opinions about feature availability, so _adapters_ should be what users import runtime-specific values/functions from. Runtime packages are an implementation detail.
+
+Goals:
+- make DX and docs simpler: more intuitive, less magic
+- reduce bugs from user error
+- reduce bugs from `remix` internal complexity
+
+Non-goals:
+- maintainability of `remix` repo
+- reduce or increase number of packages
+- more portable `app/` directory
```

Comment:
It is currently still the recommended way of changing adapters to just create a new project and copy over the `app` directory though.

https://twitter.com/mjackson/status/1532361967133831174